File: Projects/Project_Cinema Website/backend/src/main/java/com/cinema_website/backend/service/HallService.java
package com.cinema_website.backend.service;

import com.cinema_website.backend.dto.HallDTO;
import com.cinema_website.backend.mapper.HallMapper;
import com.cinema_website.backend.repository.HallRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class HallService {

    private final HallRepository hallRepository;
    private final HallMapper hallMapper;

    @Autowired
    public HallService(HallRepository hallRepository, HallMapper hallMapper) {
        this.hallRepository = hallRepository;
        this.hallMapper = hallMapper;
    }

    // GET
    public HallDTO getHall(int hallId) {
        return hallRepository.getHallById(hallId);
    }

    public List<HallDTO> getAllHalls() {
        return hallRepository.getAllHalls();
    }

    public List<HallDTO> getHallsByCinemaId(int cinemaId) {
        return hallRepository.getHallsByCinemaId(cinemaId);
    }



    // INSERT
    public boolean addHall(HallDTO hall) {
        return hallRepository.addHall(hall);
    }

    // UPDATE
    public boolean updateHall(int hallId, HallDTO updatedHall) {
        return hallRepository.updateHallById(hallId, updatedHall);
    }

    // DELETE
    public boolean deleteHall(int hallId) {
        return hallRepository.deleteHallById(hallId);
    }
}

File: Projects/Project_Cinema Website/frontend/src/services/HallService.js
import axios from "axios";

const baseURL = 'http://localhost:8080/api/halls'; // Replace with your actual API endpoint

class HallService {

    // GET 
    static async getHallById(hallId) {
        return (await axios.get(`${baseURL}/getHall/${hallId}`)).data;
    }

    static async getAllHalls() {
        return (await axios.get(`${baseURL}/getAllHalls`)).data;
    }

    static async getHallsByCinemaId(cinemaId) {
        return (await axios.get(`${baseURL}/getHallsByCinemaId/${cinemaId}`)).data;
    }

    // POST
    static async addHall(newHall) {
        return (await axios.post(`${baseURL}/addHall`, newHall)).data;
    }

    // PUT (Update)
    static async updateHallById(hallId, updatedHall) {
        return (await axios.put(`${baseURL}/updateHall/${hallId}`, updatedHall)).data;
    }

    // DELETE
    static async deleteHallById(hallId) {
        return (await axios.delete(`${baseURL}/deleteHall/${hallId}`)).data;
    }
}

export default HallService;
